### PR TITLE
Add checkbox to redirect New assets to intake

### LINKED ISSUE #454: collins crashes when accessing asset without ipmi data
Hi,

I've created a new asset with generate_ipmi=false and uploaded lldp and lshw data. All that worked but when accessing the asset via the UI (probably API as well) collins throws an "Oops, an error occured" with this in the log:

```
2016-08-16 15:41:00.118462500  [error] application - 
2016-08-16 15:41:00.118479500  
2016-08-16 15:41:00.118482500  ! @714phimm1 - Internal server error, for (GET) [/resource/76] ->
2016-08-16 15:41:00.118484500  
2016-08-16 15:41:00.118486500  play.api.Application$$anon$1: Execution exception[[IllegalStateException: No IPMI configuration for asset MY_ASSET]]
2016-08-16 15:41:00.118489500  	at play.api.Application$class.handleError(Application.scala:296) ~[com.typesafe.play.play_2.11-2.3.9.jar:2.3.9]
2016-08-16 15:41:00.118491500  	at play.api.DefaultApplication.handleError(Application.scala:402) [com.typesafe.play.play_2.11-2.3.9.jar:2.3.9]
2016-08-16 15:41:00.118493500  	at play.core.server.netty.PlayDefaultUpstreamHandler$$anonfun$14$$anonfun$apply$1.applyOrElse(PlayDefaultUpstreamHandler.scala:205) [com.typesafe.play.play_2.11-2.3.9.jar:2.3.9]
2016-08-16 15:41:00.118497500  	at play.core.server.netty.PlayDefaultUpstreamHandler$$anonfun$14$$anonfun$apply$1.applyOrElse(PlayDefaultUpstreamHandler.scala:202) [com.typesafe.play.play_2.11-2.3.9.jar:2.3.9]
2016-08-16 15:41:00.118499500  	at scala.runtime.AbstractPartialFunction.apply(AbstractPartialFunction.scala:36) [org.scala-lang.scala-library-2.11.7.jar:na]
2016-08-16 15:41:00.118501500  Caused by: java.lang.IllegalStateException: No IPMI configuration for asset MY_ASSET
2016-08-16 15:41:00.118503500  	at collins.power.management.IpmiPowerCommand$.fromPowerAction(PowerManagement.scala:56) ~[collins.collins-2.0-SNAPSHOT.jar:2.0-SNAPSHOT]
2016-08-16 15:41:00.118506500  	at collins.controllers.actions.resources.IntakeStage1Action.identifyAsset(IntakeStage1Action.scala:63) ~[collins.collins-2.0-SNAPSHOT.jar:2.0-SNAPSHOT]
2016-08-16 15:41:00.118508500  	at collins.controllers.actions.resources.IntakeStage1Action.execute(IntakeStage1Action.scala:54) ~[collins.collins-2.0-SNAPSHOT.jar:2.0-SNAPSHOT]
2016-08-16 15:41:00.118511500  	at collins.controllers.actions.SecureAction$$anonfun$handleExecution$1.apply(SecureAction.scala:173) ~[collins.collins-2.0-SNAPSHOT.jar:2.0-SNAPSHOT]
2016-08-16 15:41:00.118513500  	at collins.controllers.actions.SecureAction$$anonfun$handleExecution$1.apply(SecureAction.scala:173) ~[collins.collins-2.0-SNAPSHOT.jar:2.0-SNAPSHOT]
2016-08-16 15:41:00.119889500  [warn] application - Unhandled exception
2016-08-16 15:41:00.119906500  play.api.Application$$anon$1: Execution exception[[IllegalStateException: No IPMI configuration for asset MY_ASSET]]
2016-08-16 15:41:00.119909500  	at play.api.Application$class.handleError(Application.scala:296) ~[com.typesafe.play.play_2.11-2.3.9.jar:2.3.9]
2016-08-16 15:41:00.119911500  	at play.api.DefaultApplication.handleError(Application.scala:402) [com.typesafe.play.play_2.11-2.3.9.jar:2.3.9]
2016-08-16 15:41:00.119913500  	at play.core.server.netty.PlayDefaultUpstreamHandler$$anonfun$14$$anonfun$apply$1.applyOrElse(PlayDefaultUpstreamHandler.scala:205) [com.typesafe.play.play_2.11-2.3.9.jar:2.3.9]
2016-08-16 15:41:00.119916500  	at play.core.server.netty.PlayDefaultUpstreamHandler$$anonfun$14$$anonfun$apply$1.applyOrElse(PlayDefaultUpstreamHandler.scala:202) [com.typesafe.play.play_2.11-2.3.9.jar:2.3.9]
2016-08-16 15:41:00.119919500  	at scala.runtime.AbstractPartialFunction.apply(AbstractPartialFunction.scala:36) [org.scala-lang.scala-library-2.11.7.jar:na]
2016-08-16 15:41:00.119921500  Caused by: java.lang.IllegalStateException: No IPMI configuration for asset MY_ASSET
2016-08-16 15:41:00.119923500  	at collins.power.management.IpmiPowerCommand$.fromPowerAction(PowerManagement.scala:56) ~[collins.collins-2.0-SNAPSHOT.jar:2.0-SNAPSHOT]
2016-08-16 15:41:00.119925500  	at collins.controllers.actions.resources.IntakeStage1Action.identifyAsset(IntakeStage1Action.scala:63) ~[collins.collins-2.0-SNAPSHOT.jar:2.0-SNAPSHOT]
2016-08-16 15:41:00.119928500  	at collins.controllers.actions.resources.IntakeStage1Action.execute(IntakeStage1Action.scala:54) ~[collins.collins-2.0-SNAPSHOT.jar:2.0-SNAPSHOT]
2016-08-16 15:41:00.119930500  	at collins.controllers.actions.SecureAction$$anonfun$handleExecution$1.apply(SecureAction.scala:173) ~[collins.collins-2.0-SNAPSHOT.jar:2.0-SNAPSHOT]
2016-08-16 15:41:00.119932500  	at collins.controllers.actions.SecureAction$$anonfun$handleExecution$1.apply(SecureAction.scala:173) ~[collins.collins-2.0-SNAPSHOT.jar:2.0-SNAPSHOT]
```

I've created this system the same way I created all our other systems except the lshw data is 'handcrafted'. I can provide that but it seems unrelated. Then again, I don't know what else would case this state.

### PR DESCRIPTION
**Pull request by discordianfish**
This adds a "Intake Redirect" checkbox on the index page. When checked
and the search returns a single asset with status New, the browser get
redirected to the intake process.

This replaces the old behavior where a search for only the tag always
redirects to intake process if the asset is New.

--
Closes #454 
Related: #484

/cc @byxorna @Primer42 Would that work for you? This would mean a minor change in your workflow: Instead if just scanning the tag to get to the intake process, you would initially set the checkbox once. If you just click return, the checkbox should stay checked. If you reload, it will get unchecked. But I can look into storing the checkbox state in the browser somehow if you think this is needed.
See also discussion in #454

### CODE REVIEW

**Comment by byxorna** [comment]
This LGTM, awesome!

**Comment by roymarantz** [comment]
Instead of hardcoding the default value, perhaps get it from the configuration?  That way the default could be set for the common workflow and the check box would just override it.  I'm OK with this just coming from the config file instead of the checkbox too.

**Comment by discordianfish** [comment]
@roymarantz Which default value? Whether it's checked or not?
I don't really care much about whether to use a checkbox or config. @byxorna / @Primer42: Thoughts?

**Comment by roymarantz** [comment]
yes whether it is checked or not.

**Comment by discordianfish** [comment]
@roymarantz Like this?

File: conf/reference/features_reference.conf [hunk 1]
```diff
@@ -28,6 +28,9 @@ features {
   # Whether the tumblr intake process is supported
   intakeSupported = true
 
+  # Wether to enable redirect to intake checkbox by default
```
Comment (roymarantz) [inline conf/reference/features_reference.conf]: A couple of minor changes:
 Whether to set the Redirect to Intake checkbox by default

**Comment by discordianfish** [comment]
@roymarantz [x] done

**Comment by byxorna** [comment]
@discordianfish could you document this behavior in `gh-pages` branch? It appears there is no succinct description of the intake process at all in our docs (that i could find), so making a section in https://tumblr.github.io/collins/concepts.html would be fantastic to describe the normal flow, and how this checkbox impacts behavior.

@roymarantz can you re-review and remove your blocking review if all looks good?

File: app/views/resources/index.scala.html [hunk 2]
```diff
@@ -130,10 +131,26 @@
             }
           }
 
+          @if(Feature.intakeSupported) {
```
Comment (byxorna) [inline app/views/resources/index.scala.html]: I would like to perhaps investigate only showing this button if `controllers.Resources.intake` is enabled for the user? @discordianfish is that possible to test?

**Comment by discordianfish** [comment]
Sorry but I won't have time for this anytime soon. Maybe @davidbirdsong wants to pick that up.

**Pull request closed without merging**